authority/authority.go

```diff
@@ -115,6 +115,20 @@ func New(cfg *config.Config, opts ...Option) (*Authority, error) {
 	return a, nil
 }
 
+// FromOptions creates an Authority exclusively using the passed in options
+// and does not initialize the Authority.
+func FromOptions(opts ...Option) (*Authority, error) {
+	var a = new(Authority)
+
+	// Apply options.
+	for _, fn := range opts {
+		if err := fn(a); err != nil {
+			return nil, err
+		}
+	}
+	return a, nil
+}
```

Comment:
I don't think we should expose this method, you can just copy this code in the cli and use an internal with this code.